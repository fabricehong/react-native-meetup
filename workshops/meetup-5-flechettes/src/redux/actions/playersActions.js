export const ADD_PLAYER = 'ADD_PLAYER';
export const REMOVE_PLAYER = 'REMOVE_PLAYER';
export const SET_PLAYER_NAME = 'SET_PLAYER_NAME';
export const SET_PLAYER_SCORE = 'SET_PLAYER_SCORE';

export function addPlayer(name) {
  return {
    type: ADD_PLAYER, name: name,
  };
}

export function removePlayer(id) {
  return {
    type: REMOVE_PLAYER, id: id,
  };
}

export function setPlayerName(name, id) {
  return {
    type: SET_PLAYER_NAME, id: id, name:name
  };
}
