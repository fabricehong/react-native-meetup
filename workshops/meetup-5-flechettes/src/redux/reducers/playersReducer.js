import {ADD_PLAYER, REMOVE_PLAYER} from '../actions/playersActions';

const defaultState = {
  players: [],
};

const playersReducer = (state = defaultState, action) => {

  const players = state.players;

  switch (action.type) {
    case ADD_PLAYER:
      players.push({name: action.name, score: 0});
      return {...state, players: players};
    case REMOVE_PLAYER:
      return {...state, players: players.filter((v, i) => i !== action.id)};
    default:
      return state;
  }
};

export default playersReducer;