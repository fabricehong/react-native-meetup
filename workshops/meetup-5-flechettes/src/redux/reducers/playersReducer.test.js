import * as actions from '../actions/playersActions';
import playersReducer from './playersReducer';

const defaultState = {
  players: []
};

describe('playersReducer', () => {

  test('Add player action should add a player to the store', () => {
    const newState = playersReducer(defaultState, actions.addPlayer('Jules'));
    expect(newState).toMatchSnapshot();
  });

});