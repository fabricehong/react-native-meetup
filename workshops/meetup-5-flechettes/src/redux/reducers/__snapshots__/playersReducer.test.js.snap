// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playersReducer Add player action should add a player to the store 1`] = `
Object {
  "players": Array [
    "Jules",
  ],
}
`;
